use version number instead of 'master' in html header title

glaringlee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38149 use version number instead of 'master' in html header title**

This is for (#21290) (#31894)

Instead of putting "Pytorch master documentation" in header's html title, now we use "Pytorch 1.x.x documentation", this is similar to tensorFlow and numpy doc page.

In google search, we will get 
Pytorch Documentation - Pytorch 1.x.x Documentation instead.

Differential Revision: [D21586559](https://our.internmc.facebook.com/intern/diff/D21586559)

glaringlee committed:
use version number instead of 'master' in html header title

[ghstack-poisoned]

zou3519 commented on docs/source/conf.py:
```diff
@@ -103,6 +103,14 @@
+# Default is " ".join(project, release, "documentation") if not set
+if RELEASE:
+    html_title = " ".join((project, torch.__version__, "documentation"))    
+else:
```
We should find some way to mark the master documentation as "not ready for general consumption". Maybe it would be better if the master documentation is still called "PyTorch master documentation" ?

glaringlee replied:
keep master as master, stable version will use 1.x.x. Remove the hash code (start with 'a') if any. eg. 1.5.0a4b+xxx -> 1.5.0

zou3519 commented on docs/source/conf.py:
```diff
@@ -103,6 +103,14 @@
+# Customized html_title here. 
+# Default is " ".join(project, release, "documentation") if not set
+if RELEASE:
+    html_title = " ".join((project, torch.__version__, "documentation"))    
```
Could you comment with a screenshot of the documentation as a quick test?

glaringlee replied:
This is in header. It changed to 
<title>PyTorch documentation — PyTorch 1.6.0 documentation</title>
now

zou3519 replied:
I was asking for a screenshot of the actual webpage that says that, as a sanity check that this code gets threaded all the way through, but I believe that it works :)

glaringlee replied:
I see. Can you review this first https://github.com/pytorch/pytorch/pull/38119.
So I can empty my github.io to put another demo there.

zou3519 replied:
Sounds good, I'll go review that first. If you have a local version of the build, you can open the html file directly on your computer and then take a screenshot of that.

zou3519 commented:
NB: this doesn't actually fix what older versions of the documentation say (like 0.4.0). To fix those we'd probably have to send PRs to https://github.com/pytorch/pytorch.github.io that modify the raw html.

However, the older documentation isn't indexed by google, so it is not the biggest problem if we don't change them.

glaringlee committed:
Update on "use version number instead of 'master' in html header title"


This is for (#21290) (#31894)

Instead of putting "Pytorch master documentation" in header's html title, now we use "Pytorch 1.x.x documentation", this is similar to tensorFlow and numpy doc page.

In google search, we will get 
Pytorch Documentation - Pytorch 1.x.x Documentation instead.

[ghstack-poisoned]

glaringlee commented:
Yes, I searched on google, it seems google does not index the old documents. 
The only thing is should we update current stable version documents?

zou3519 commented:
> Yes, the only thing is should we update current stable version documents?

Yeah, let's update just the current stable version docs.

glaringlee commented:
@zou3519 yep, will do.

glaringlee committed:
Update on "use version number instead of 'master' in html header title"


This is for (#21290) (#31894)

Instead of putting "Pytorch master documentation" in header's html title, now we use "Pytorch 1.x.x documentation", this is similar to tensorFlow and numpy doc page.

In google search, we will get 
Pytorch Documentation - Pytorch 1.x.x Documentation instead.

[ghstack-poisoned]

zou3519 reviewed (approved):
This lgtm, but I would like to see a screenshot of it actually working by building the docs by using `make html-stable`

glaringlee commented:
@zou3519 Sure. I will have it ready for you to take a loot at before landing this pr.

glaringlee committed:
Update on "use version number instead of 'master' in html header title"


This is for (#21290) (#31894)

Instead of putting "Pytorch master documentation" in header's html title, now we use "Pytorch 1.x.x documentation", this is similar to tensorFlow and numpy doc page.

In google search, we will get 
Pytorch Documentation - Pytorch 1.x.x Documentation instead.

[ghstack-poisoned]

glaringlee commented:
@zou3519 
Demo is available now at https://glaringlee.github.io/
Check the header's title in any pages' html, 'pytorch master documentation' is replaced by 'pytorch 1.6.0 documentation'

Commit:
Update on "use version number instead of 'master' in html header title"


This is for (#21290) (#31894)

Instead of putting "Pytorch master documentation" in header's html title, now we use "Pytorch 1.x.x documentation", this is similar to tensorFlow and numpy doc page.

In google search, we will get 
Pytorch Documentation - Pytorch 1.x.x Documentation instead.

Differential Revision: [D21586559](https://our.internmc.facebook.com/intern/diff/D21586559)

[ghstack-poisoned]

glaringlee committed:
Update on "use version number instead of 'master' in html header title"


This is for (#21290) (#31894)

Instead of putting "Pytorch master documentation" in header's html title, now we use "Pytorch 1.x.x documentation", this is similar to tensorFlow and numpy doc page.

In google search, we will get 
Pytorch Documentation - Pytorch 1.x.x Documentation instead.

Differential Revision: [D21586559](https://our.internmc.facebook.com/intern/diff/D21586559)

[ghstack-poisoned]

glaringlee committed:
Update on "use version number instead of 'master' in html header title"


This is for (#21290) (#31894)

Instead of putting "Pytorch master documentation" in header's html title, now we use "Pytorch 1.x.x documentation", this is similar to tensorFlow and numpy doc page.

In google search, we will get 
Pytorch Documentation - Pytorch 1.x.x Documentation instead.

Differential Revision: [D21586559](https://our.internmc.facebook.com/intern/diff/D21586559)

[ghstack-poisoned]
